script/generate_needle_previews

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/perl -w
+# Copyright 2025 Roland Clobus <rclobus@rclobus.nl>
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+
+=head1 generate_needle_previews
+
+generate_needle_previews - generate .svg files with the needle information as an overlay
+
+=head1 SYNOPSIS
+
+cd products/XXX/needles
+ls -1 "*.json" | generate_needle_previews
+
+=cut
+
+use v5.10;
+use strict;
+use warnings;
+use autodie;
+use Image::PNG;
+use JSON qw( decode_json );
+use Data::Dumper;
+
+sub process_file {
+    my ($filename) = @_;
+    my $FILE;
+
+    chomp $filename;
+    my $filename_without_extension = substr($filename, 0, length($filename) - 5);
+    printf "Working on %s\n", $filename_without_extension;
+    my $png = Image::PNG->new();
+    $png->read($filename_without_extension . '.png');
+
+    my $json_text = do {
+        open(my $json_fh, '<:encoding(UTF-8)', $filename_without_extension . '.json');
+        local $/;
+        <$json_fh>;
+    };
+    my $data  = decode_json $json_text;
+    my @areas = @{$data->{'area'}};
+
+    open $FILE, '>', $filename_without_extension . '.svg';
+    select $FILE;
+    say '<?xml version="1.0" encoding="UTF-8" standalone="no"?>';
+    say '<!-- Created with generate_needle_previews.pl -->';
+    say '<svg';
+    say '   version="1.1"';
+    say '   id="svg2"';
+    printf "   width=\"%d\"\n",          $png->width;
+    printf "   height=\"%d\"\n",         $png->height;
+    printf "   viewBox=\"0 0 %d %d\"\n", $png->width, $png->height;
+    say '   xmlns:xlink="http://www.w3.org/1999/xlink"';
+    say '   xmlns="http://www.w3.org/2000/svg"';
+    say '   xmlns:svg="http://www.w3.org/2000/svg">';
+    say '  <defs';
+    say '     id="defs6" />';
+    say '  <g';
+    say '     id="g8">';
+    say '    <image';
+    printf "       width=\"%d\"\n",  $png->width;
+    printf "       height=\"%d\"\n", $png->height;
+    say '       preserveAspectRatio="none"';
+    printf "       xlink:href=\"%s\"\n", $filename_without_extension . '.png';
+    say '       id="needle_bitmap" />';
+    foreach my $area (@areas) {
+        say '    <rect';
+        if ($area->{'type'} eq 'match') {
+            say '       style="fill:#00ff00;fill-opacity:0.5;stroke:#008000;stroke-width:2"';
+        } elsif ($area->{'type'} eq 'exclude') {
+            say '       style="fill:#ff0000;fill-opacity:0.5;stroke:#800000;stroke-width:2"';
+        } else {
+            say '       style="fill:#000000;fill-opacity:0.5;stroke:#808080;stroke-width:2"';
+        }
+        say '       id="needle001"';
+        printf "       width=\"%d\"",  $area->{'width'};
+        printf "       height=\"%d\"", $area->{'height'};
+        printf "       x=\"%d\"",      $area->{'xpos'};
+        printf "       y=\"%d\" />",   $area->{'ypos'};
+    }
+    say '  </g>';
+    say '</svg>';
+    select STDOUT;
+    close $FILE;
+}
+
+# Read a list of files to process
+while (<STDIN>) {
```

Comment:
Maybe we can make that optional and only process STDIN if the user specifies `-`? Other opinions.

In any case, this needs a `--help` output, as `t/44-scripts.t` checks for that.
You can probably just copy from another simple script in our directory.
https://github.com/os-autoinst/os-autoinst/blob/master/script/check_qemu_oom#L20-L28